components/inputs/input-text.js

```diff
@@ -56,20 +81,73 @@ class InputText extends RtlMixin(LitElement) {
 		this.preventSubmit = false;
 		this.readonly = false;
 		this.required = false;
+		this.slotPaddingWidth = 2.2;
 		this.type = 'text';
 		this.value = '';
+
+		this._focused = false;
+		this._hasSlot = false;
+		this._hovered = false;
+		this._slotElems = [];
+		this._padding = [];
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		const slots = this.shadowRoot.querySelectorAll('slot');
+		slots.forEach((slot) => slot.addEventListener('slotchange', this._onSlotChange.bind(this)));
+
+		this.addEventListener('blur', this._handleBlur);
+		this.addEventListener('focus', this._handleFocus);
+
```

Comment:
Could do this with inline events on the slots themselves: `<slot @slotchange="${this._handleBlur}">`